File: python-client-sdk/src/orbapi/models/operations/post_link_token.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from orbapi import utils
from ..shared import security as shared_security


@dataclasses.dataclass
class PostLinkTokenSecurity:
    vessel_api_token: shared_security.SchemeVesselAPIToken = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    

@dataclass_json
@dataclasses.dataclass
class PostLinkTokenResponseBody:
    link_token: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('linkToken') }})
    

@dataclasses.dataclass
class PostLinkTokenRequest:
    security: PostLinkTokenSecurity = dataclasses.field()
    

@dataclasses.dataclass
class PostLinkTokenResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    response_body: Optional[PostLinkTokenResponseBody] = dataclasses.field(default=None)
    

File: typescript-client-sdk/src/sdk/models/operations/postlinktoken.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PostLinkTokenSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  vesselAPIToken: shared.SchemeVesselApiToken;
}


export class PostLinkTokenResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=linkToken" })
  linkToken?: string;
}


export class PostLinkTokenRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  security: PostLinkTokenSecurity;
}


export class PostLinkTokenResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: PostLinkTokenResponseBody;
}
